imagebuildah/build_linux.go

```diff
@@ -0,0 +1,86 @@
+package imagebuildah
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"github.com/containers/buildah/define"
+	"github.com/containers/buildah/internal/tmpdir"
+	"github.com/containers/buildah/pkg/overlay"
+	"github.com/containers/storage"
+	"github.com/opencontainers/selinux/go-selinux/label"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+// platformSetupContextDirectoryOverlay() may set up an overlay _over_ the
+// build context directory, and sorts out labeling.  Returns either the new
+// location which should be used as the base build context or the old location;
+// the process label and mount label for the build; a boolean value that
+// indicates whether we did, in fact, mount an overlay; a cleanup function
+// which should be called when the location is no longer needed (on success);
+// and a non-nil fatal error if any of that failed.
+func platformSetupContextDirectoryOverlay(store storage.Store, options *define.BuildOptions) (string, string, string, bool, func(), error) {
```

Comment:
Return values are a bit unwieldy, maybe a `struct` instead? Although I guess the single caller unpacks them anyways, so doesn't matter. Yeah, one caller, so makes sense as is.